core/templates/dev/head/services/AssetsBackendApiService.js

```diff
@@ -58,33 +68,56 @@ oppia.factory('AssetsBackendApiService', [
           if (exception.name === 'TypeError' && window.BlobBuilder) {
             var blobBuilder = new BlobBuilder();
             blobBuilder.append(data);
-            var audioBlob = blobBuilder.getBlob('audio/*');
+            var assetBlob = blobBuilder.getBlob(assetType.concat('/*'));
           } else {
             throw exception;
           }
         }
-        assetsCache[filename] = audioBlob;
-        successCallback(AudioFileObjectFactory.createNew(filename, audioBlob));
+        assetsCache[filename] = assetBlob;
+        if (assetType === 'audio') {
+          successCallback(
+            AudioFileObjectFactory.createNew(filename, assetBlob));
+        } else {
+          successCallback(
+            ImageFileObjectFactory.createNew(filename, assetBlob));
+        }
       }).error(function() {
         errorCallback();
       })['finally'](function() {
-        _removeFromFilesCurrentlyBeingRequested(filename);
+        _removeFromFilesCurrentlyBeingRequested(filename, assetType);
       });
     };
 
-    var _abortAllCurrentDownloads = function() {
-      _filesCurrentlyBeingRequested.forEach(function(request) {
-        request.canceler.resolve();
-      });
-      _filesCurrentlyBeingRequested = [];
+    var _abortAllCurrentDownloads = function(assetType) {
+      if (assetType === 'audio') {
```

Comment:
`'audio'` and `'image'` should probably be constants.

Reply:
Done!